### DISCUSSION TITLE
Problem with code

### DISCUSSION THREAD

**Discussion by murxg** [community/community#22488] [Other Feature Feedback, Questions, & Ideas]
<p>I have problem with my code:</p>
<pre><code class="lang-auto">&lt;!DOCTYPE html&gt;
&lt;html&gt;
&lt;head&gt;
&lt;title&gt;Snake&lt;/title&gt;

      &lt;meta property="og:image" content="https://cdn-icons-png.flaticon.com/512/427/427533.png"&gt;
      &lt;meta name = "keywords" content = "HTML, snake, NPCFISH, online, free" /&gt;
      &lt;meta name = "description" content = "Simple snake!" /&gt;
      &lt;meta name = "author" content = "NPCFISH" /&gt;
      &lt;meta http-equiv = "Content-Type" content = "text/html; charset = UTF-8" /&gt;

&lt;link rel="icon" href="https://cdn-icons-png.flaticon.com/512/427/427533.png"&gt;
&lt;meta charset="UTF-8"&gt;
&lt;style&gt;
html, body {
height: 100%;
margin: 0;
  }
body {
background: black;
display: flex;
align-items: center;
justify-content: center;
  }
canvas {
border: 1px solid white;
  }
&lt;/style&gt;
&lt;/head&gt;
&lt;body&gt;
&lt;canvas width=1500 height=700 id="game"&gt;&lt;/canvas&gt;
&lt;script&gt;
var canvas = document.getElementById('game');
var context = canvas.getContext('2d');
var grid = 16;
var count = 0;
var snake = {
x: 160,
y: 160,
dx: grid,
dy: 0,
cells: [],
maxCells: 4
};
var apple = {
x: 320,
y: 320
};
function getRandomInt(min, max) {
return Math.floor(Math.random() * (max - min)) + min;
}
// game loop
function loop() {
requestAnimationFrame(loop);
if (++count &lt; 4) {
return;
}
count = 0;
context.clearRect(0,0,canvas.width,canvas.height);
snake.x += snake.dx;
snake.y += snake.dy;
if (snake.x &lt; 0) {
snake.x = canvas.width - grid;
}
else if (snake.x &gt;= canvas.width) {
snake.x = 0;
}
if (snake.y &lt; 0) {
snake.y = canvas.height - grid;
}
else if (snake.y &gt;= canvas.height) {
snake.y = 0;
}
snake.cells.unshift({x: snake.x, y: snake.y});
if (snake.cells.length &gt; snake.maxCells) {
snake.cells.pop();
}
// draw apple
context.fillStyle = 'red';
context.fillRect(apple.x, apple.y, grid-1, grid-1);
// draw snake one cell at a time
context.fillStyle = 'green';
snake.cells.forEach(function(cell, index) {
context.fillRect(cell.x, cell.y, grid-1, grid-1);
// snake ate apple
if (cell.x === apple.x &amp;&amp; cell.y === apple.y) {
snake.maxCells++;
apple.x = getRandomInt(0, 25) * grid;
apple.y = getRandomInt(0, 25) * grid;
}
for (var i = index + 1; i &lt; snake.cells.length; i++) {
if (cell.x === snake.cells[i].x &amp;&amp; cell.y === snake.cells[i].y) {
snake.x = 160;
snake.y = 160;
snake.cells = [];
snake.maxCells = 4;
snake.dx = grid;
snake.dy = 0;
apple.x = getRandomInt(0, 25) * grid;
apple.y = getRandomInt(0, 25) * grid;
}
}
});
}
// listen to keyboard events to move the snake
document.addEventListener('keydown', function(e) {
// left arrow key
if (e.which === 37 &amp;&amp; snake.dx === 0) {
snake.dx = -grid;
snake.dy = 0;
}
// up arrow key
else if (e.which === 38 &amp;&amp; snake.dy === 0) {
snake.dy = -grid;
snake.dx = 0;
}
// right arrow key
else if (e.which === 39 &amp;&amp; snake.dx === 0) {
snake.dx = grid;
snake.dy = 0;
}
// down arrow key
else if (e.which === 40 &amp;&amp; snake.dy === 0) {
snake.dy = grid;
snake.dx = 0;
}
});
// start the game
requestAnimationFrame(loop);
&lt;/script&gt;
&lt;/body&gt;
&lt;/html&gt;
</code></pre>
<p>I tried to make <code>&lt;canvas&gt;</code> so it would be same size as the window. Can someone help me? :pleading_face:</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group" data-username="TheNPCFISH" data-post="1" data-topic="230526">
<div class="title">
<div class="quote-controls"></div>
 TheNPCFISH:</div>
<blockquote>
<p>I tried to make <code>&lt;canvas&gt;</code> so it would be same size as the window. Can someone help me?</p>
</blockquote>
</aside>
<p>You’re providing a fix predefined size to the <code>&lt;canvas&gt;</code> element:</p>
<pre data-code-wrap="html"><code class="lang-nohighlight">&lt;canvas width=1500 height=700 id="game"&gt;&lt;/canvas&gt;
</code></pre>
<p>you might want to start by using percentage values in either <code>width</code> or <code>height</code>, and maybe auto-adjust the other value — or just use <code>100%</code> on both. The problem is that you’d then end up with a varying-size canvas, is that going to be OK for what you’re planning to do?</p>
<p>That of the “window” is a bit of a slippery concept inside the browser, since the end user can freely resize the main browser window, thus changing the size of available “window” to the actual HTML page (i.e. excluding the toolbar, and other surrounding elements).</p>
<p>Probably the best approach would be to dynamically detect the available window size using JavaScript, and detect any changes to the windows size while your app is running.</p>
